hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneDebugReplicasVerify.java

```diff
@@ -0,0 +1,370 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Stream;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.hadoop.hdds.client.ReplicationConfig;
+import org.apache.hadoop.hdds.client.ReplicationFactor;
+import org.apache.hadoop.hdds.client.ReplicationType;
+import org.apache.hadoop.hdds.scm.ScmConfigKeys;
+import org.apache.hadoop.ozone.HddsDatanodeService;
+import org.apache.hadoop.ozone.TestDataUtil;
+import org.apache.hadoop.ozone.client.ObjectStore;
+import org.apache.hadoop.ozone.client.OzoneBucket;
+import org.apache.hadoop.ozone.client.OzoneKey;
+import org.apache.hadoop.ozone.client.OzoneVolume;
+import org.apache.hadoop.ozone.container.common.interfaces.Container;
+import org.apache.hadoop.ozone.debug.OzoneDebug;
+import org.apache.hadoop.ozone.om.OMConfigKeys;
+import org.apache.hadoop.ozone.om.helpers.OmKeyArgs;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfoGroup;
+import org.apache.ozone.test.NonHATests;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Test Ozone Debug shell.
+ */
+public abstract class TestOzoneDebugReplicasVerify implements NonHATests.TestCase {
+  private static final Logger LOG = LoggerFactory.getLogger(TestOzoneDebugReplicasVerify.class);
+  private static final String CHUNKS_DIR_NAME = "chunks";
+  private static final String BLOCK_FILE_EXTENSION = ".block";
+
+  @TempDir
+  private File tempDir;
+  private OzoneDebug ozoneDebugShell;
+  private String volume;
+  private String bucket;
+  private String key;
+  private String ozoneAddress;
+  private StringWriter err;
+  private StringWriter out;
+
+  @BeforeEach
+  void init() {
+    out = new StringWriter();
+    err = new StringWriter();
+    ozoneDebugShell = new OzoneDebug();
+    ozoneDebugShell.getCmd().setOut(new PrintWriter(out));
+    ozoneDebugShell.getCmd().setErr(new PrintWriter(err));
+  }
+
+  @BeforeEach
+  void setupKeys() throws IOException {
+    volume = generateVolume("vol-a-");
+    ozoneAddress = "/" + volume;
+    generateKeys(volume, "level1/multiLevelKey-", 10);
+    generateKeys(volume, "key-", 10);
+  }
+
+  @AfterEach
+  void cleanupKeys() throws IOException {
+    if (!out.toString().isEmpty()) {
+      LOG.info(out.toString());
+    }
+
+    if (!err.toString().isEmpty()) {
+      LOG.error(err.toString());
+    }
+
+    ObjectStore store = cluster().newClient().getObjectStore();
```

Comment:
Whenever you create a `newClient()`, make sure to close it as well.  Either store a single client as member variable and close in `@AfterEach`, or use it in `try-with-resources` block.